### PR TITLE
Check before calling mkdirs on staging directory

### PR DESCRIPTION
**Pull request by kalman**
Also run pylint on staging.py.

### CODE REVIEW

File: tools/noms/staging.py [hunk 1]
```diff
@@ -4,66 +4,75 @@
 # Licensed under the Apache License, version 2.0:
 # http://www.apache.org/licenses/LICENSE-2.0
 
-import argparse, glob, shutil, os, os.path
+import argparse
+import glob
+import os
+import os.path
+import shutil
```
Comment (kalman) [inline tools/noms/staging.py]: pylint doesn't like multiple imports on same line

File: tools/noms/staging.py [hunk 2]
```diff
@@ -4,66 +4,75 @@
...
 
 def Main(projectName, stagingFunction):
-	"""Main should be called by all staging scripts when executed.
+    """Main should be called by all staging scripts when executed.
```
Comment (kalman) [inline tools/noms/staging.py]: pylint doesn't like tabs, and neither does Python.

File: tools/noms/staging.py [hunk 3]
```diff
@@ -4,66 +4,75 @@
...
-	normalized = os.path.realpath(projectStagingDir)
-	if not _isSubDir(projectStagingDir, args.stagingDir):
-		raise Exception(projectStagingDir + ' must be a subdir of ' + args.stagingDir)
+    normalized = os.path.realpath(projectStagingDir)
+    if not _is_sub_dir(projectStagingDir, args.stagingDir):
+        raise Exception(projectStagingDir + ' must be a subdir of ' + args.stagingDir)
 
-	os.makedirs(normalized)
-	stagingFunction(normalized)
+    if not os.path.exists(normalized):
```
Comment (kalman) [inline tools/noms/staging.py]: This is the only actual functional change, causing deploy to fail

File: tools/noms/staging.py [hunk 4]
```diff
@@ -4,66 +4,75 @@
...
+        raise ValueError(arg + ' is not a path to a directory.')
+    return normalized
+
+
+def _is_sub_dir(subdir, directory):
+    # Need the path-sep at the end to ensure that commonprefix returns the correct result below.
+    directory = os.path.join(os.path.realpath(directory), '')
+    subdir = os.path.realpath(subdir)
+
+    # return true, if the common prefix of both is equal to directory e.g.  /a/b/c/d.rst and
```
Comment (kalman) [inline tools/noms/staging.py]: pylint doesn't like really long lines

**Comment by arv** [comment]
LGTM

**Pull request merged**